id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java

```diff
@@ -622,20 +659,25 @@ private byte[] getBiometricsForRequestedFormats(String uinHash, String fileName,
 		}
 	}
 
+	
+	public IdResponseDTO updateIdentity(IdRequestDTO request, String uin) throws IdRepoAppException {
+		return updateIdentity(request, uin, false);
+	}
+
 	/*
 	 * (non-Javadoc)
 	 * 
 	 * @see io.mosip.kernel.core.idrepo.spi.IdRepoService#updateIdentity(java.lang.
 	 * Object, java.lang.String)
 	 */
 	@Override
-	public IdResponseDTO updateIdentity(IdRequestDTO request, String uin) throws IdRepoAppException {
+	public IdResponseDTO updateIdentity(IdRequestDTO request, String uin,boolean overridedata) throws IdRepoAppException {

```

Comment:
As per this change the notify method will be called with isUpdate as true for a addIdenity, will this cause any issues ? Can you check on this with @LoganathanSekar7627

Reply:
No, it will not cause issue as we are calling the Update method in add identity on the basis of the condition. I checked with @LoganathanSekar7627  as well.

Reply:
My concern is the rid is for add identity but there is this below behaviour change on update
https://github.com/mosip/id-repository/blob/34c4fd20a8c87878345b3a5d4b4ee864c9d5c183/id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java#L746

Also we call the update on the next service layer, we have validated the logic and it goes do something we don't want on add identity ?

Reply:
As discussed with loga  it is clear now.

Reply:
@LoganathanSekar7627 Reviewed the code and cleared that there will be no impact using the update method and @Neha2365 tested and confirmed that all the user attributes are getting updates as expected, so resolved.